store/postgres/src/chain_store.rs

```diff
@@ -1028,6 +1028,69 @@ mod data {
                 .collect())
         }
 
+        pub(super) fn clear_call_cache(
+            &self,
+            conn: &PgConnection,
+            from: Option<i32>,
+            to: Option<i32>,
+        ) -> Result<usize, Error> {
+            match self {
+                Storage::Shared => {
+                    use public::eth_call_cache as cache;
+                    let delete_stmt = diesel::delete(cache::table);
+                    match (from, to) {
+                        (Some(from), None) => delete_stmt
+                            .filter(cache::block_number.ge(from))
+                            .execute(conn)
+                            .map_err(Error::from),
+                        (None, Some(to)) => delete_stmt
+                            .filter(cache::block_number.le(to))
+                            .execute(conn)
+                            .map_err(Error::from),
+                        (Some(from), Some(to)) => delete_stmt
+                            .filter(cache::block_number.ge(from))
+                            .filter(cache::block_number.le(to))
+                            .execute(conn)
+                            .map_err(Error::from),
+                        (None, None) => delete_stmt.execute(conn).map_err(Error::from),
+                    }
+                }
+                Storage::Private(Schema { call_cache, .. }) => match (from, to) {
+                    (Some(from), None) => {
+                        let query =
+                            format!("delete from {} where block_number >= $1", call_cache.qname);
+                        sql_query(query)
+                            .bind::<Integer, _>(from)
+                            .execute(conn)
+                            .map_err(Error::from)
+                    }
+                    (None, Some(to)) => {
+                        let query =
+                            format!("delete from {} where block_number <= $1", call_cache.qname);
+                        sql_query(query)
+                            .bind::<Integer, _>(to)
+                            .execute(conn)
+                            .map_err(Error::from)
+                    }
+                    (Some(from), Some(to)) => {
+                        let query = format!(
+                            "delete from {} where block_number >= $1 and block_number <= $2",
+                            call_cache.qname
+                        );
+                        sql_query(query)
+                            .bind::<Integer, _>(from)
+                            .bind::<Integer, _>(to)
+                            .execute(conn)
+                            .map_err(Error::from)
+                    }
```

Comment:
We would instead use Diesel methods than plain SQL here, just like you did for the `Store::Shared` variant.

Please refer to my comment over there, as this scope could also benefit from removing the code duplication and using `anyhow::Context`.

Reply:
You can find the value for the call cache table at `call_cache.table`.
Here's an example of that table being used as a Diesel table type:
https://github.com/graphprotocol/graph-node/blob/fed567c028702888683ee801633e81f4c489e7dd/store/postgres/src/chain_store.rs#L1008-L1017

Reply:
It seems that we can't use our call cache Diesel tables for deletions 😓.
They are of type [`diesel_dynamic_schema::Table`](https://docs.rs/diesel-dynamic-schema/0.2.0/diesel_dynamic_schema/struct.Table.html) and don't implement the [`HasTable`](https://docs.rs/diesel/latest/diesel/associations/trait.HasTable.html) trait, which is required for deletion.

Reply:
Thanks for verifying it for me. I’ll resolve the comments